php/src/Google/Protobuf/Internal/MapEntry.php

```diff
@@ -32,13 +32,27 @@
 
 namespace Google\Protobuf\Internal;
 
+use Google\Protobuf\Internal\GPBType;
 use Google\Protobuf\Internal\Message;
 
 class MapEntry extends Message
 {
     public $key;
     public $value;
 
+    public function __construct($desc) {
+        parent::__construct($desc);
+        // For MapEntry, getValue should always return a valid value. Thus, we
+        // need to create a default instance value if the value type is
+        // message, in case no value is provided in data.
+        $value_field = $desc->getFieldByNumber(2);
```

Comment:
why the number 2 here specifically?

Reply:
Map entry only has key (1) and value (2) field.
Besides, only value field can be message type.